odata-csdl/14 Vocabulary and Annotation.md

```diff
@@ -1364,49 +1364,149 @@ Addresses/-1/Street
+target, unless that target is another annotation, in which case it is hosted
+by that annotation's target, and so on.
+
+If the value of an annotation is expressed dynamically with a path
+expression, the path evaluation rules for this expression depend on the model
+element by which the annotation is hosted. More precisely, this subsection defines:
+- An annotation embedded within or targeting another annotation or a
+  collection, record or property value of another annotation is hosted by the
+  same model element as that other annotation.
+- An annotation embedded within or targeting another kind of model element is
+  hosted by that model element.
+
+For annotations hosted by an entity container, the path is evaluated starting
+at the entity container, i.e. an empty path resolves to the entity container,
+and non-empty paths MUST start with a segment identifying a container child
+(entity set, function import, action import, or singleton). The subsequent
+segments follow the rules for path expressions targeting the corresponding
+child element.
+
+For annotations hosted by an entity set or a singleton, the path is evaluated
+starting at the entity set or singleton, i.e. an empty path resolves to the
+entity set or singleton, and non-empty paths MUST follow the rules for
+annotations targeting the declared entity type of the entity set or singleton.
+
+For annotations hosted by an entity type or complex type, the path is
+evaluated starting at the type, i.e. an empty path resolves to the type, and
+the first segment of a non-empty path MUST be a structural or navigation
+property of the type, a [type cast](#TypeCast), or a [term cast](#TermCast).
+
+For annotations hosted by an action, action import, function, function
+import, parameter, or return type, the first segment of the path MUST be a
+parameter name or `$ReturnType`.
+
+For the remaining cases, the path evaluation rules additionally depend on the
+*mode* ("container", "embedded", or "targeting") in which the hosting happens:
+- An annotation with external targeting whose target path starts with an
+  entity container is hosted in container mode.
+- An annotation with external targeting whose target path starts with another
+  kind of schema child is hosted in targeting mode.
+- An annotation embedded within another annotation or a collection, record or
+  property value of another annotation is hosted in the same mode as that
+  other annotation.
+- An annotation embedded within another kind of model element is hosted in
+  embedded mode.
+
+For annotations hosted by a structural or navigation property in container
+mode, the path is evaluated starting at the declared type of the property. An
+empty path resolves to the declared type of the property, and non-empty paths
+MUST follow the rules for annotations targeting the declared type of the
+property. If the type is primitive, the first segment of a non-empty path MUST
+be a [type cast](#TypeCast), or a [term cast](#TermCast).
```

Comment:
```suggestion
be a [type cast](#TypeCast) or a [term cast](#TermCast).
```
No comma here because "or" only separates two items, see https://livexp.com/blog/when-should-i-use-a-comma-before-or/